File: 1463-the-k-weakest-rows-in-a-matrix/1463-the-k-weakest-rows-in-a-matrix.java
class RowPower{
    int power, index;
    public RowPower(int power, int index) {
        this.power = power;
        this.index = index;
    }
}
class Solution {
    public int[] kWeakestRows(int[][] mat, int k) {
        RowPower[] rowPower= new RowPower[mat.length];
        for(int i = 0; i < mat.length; i++){
            int count = 0;
            for(int j = 0; j < mat[0].length; j++){
                count += mat[i][j];
            }
            rowPower[i] = new RowPower(count * 100 + i, i);
        }
        
        int[] ans = new int[k];
        Arrays.sort(rowPower, new Comparator<RowPower>(){
            public int compare(RowPower a, RowPower b){
                return a.power - b.power;
            }
        });
        for(int i = 0; i < k; i++){
            ans[i] = rowPower[i].index;
        }
        return ans;
    }
}

File: 1463-the-k-weakest-rows-in-a-matrix/1463-the-k-weakest-rows-in-a-matrix.py
class Solution:
    def kWeakestRows(self, mat: List[List[int]], k: int) -> List[int]:
        row_power = []
        for i in range(len(mat)):
            row_power.append({"power": mat[i].count(1)*100 + i, "index": i})
        row_power.sort(key=lambda x: x["power"])
        print(row_power)
        ans = [-1] * k
        for i in range(k):
            ans[i] = row_power[i]["index"]
        return ans